pkg/utils/hashlib/common.go

```diff
@@ -0,0 +1,41 @@
+package hashlib
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// BytesOfBytesKeccak will compute a keccak256 hash of the provided bytes of bytes slice
+func BytesOfBytesKeccak(b [][]byte) ([32]byte, error) {
+	if len(b) == 0 {
+		return [32]byte{}, nil
+	}
+
+	encodedArr, err := encodeBytesOfBytes(b)
+	if err != nil {
+		return [32]byte{}, err
+	}
+
+	return keccak256Fixed(encodedArr), nil
+}
+
+// encodeBytesOfBytes encodes the nested byte arrays into a single byte array as follows
+//  1. total number of nested arrays is encoded into fix-size 8 bytes at the front of the result
+//  2. for each nested array
+//     encode the array length into fixed-size 8 bytes, append to result
+//     append the array contents to result
+func encodeBytesOfBytes(b [][]byte) ([]byte, error) {
```

Comment:
This func seems more interesting (and flexible) than `BytesOfBytesKeccak`. If this were exported, could `BytesOfBytesKeccak` be removed in favor of callers invoking the hash themselves, after calling this func?